### PR TITLE
Infra Password and SQL Password

### PR DESCRIPTION
**Pull request by LiliamLeme**
# Type of PR
- Code changes

1 - Deploy deploy_infrastructure
- Simplify the logic - If the character is Hypen
- replace by R. Instead of a retry logic

2 - Init_enviromnent:
- Added more complex string concatenated to
- SQL password

## Purpose

1 - Deploy deploy_infrastructure
- Simplify the logic - If the character is Hypen
- replace by R. Instead of a retry logic

2 - Init_enviromnent:
- Added more complex string concatenated to
- SQL password

## Author pre-publish checklist

- [ X] Added test to prove my fix is effective or new feature works
- [ X] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

Testes log result example:
+ sp_stor_pass='Validating...-fm8....
Invalid password. The first character cannot be a hyphen. Replacing...
Rfm8....
Password Validated
+ echo 'ADLS - Valid password obtained'

## Issues Closed or Referenced

- Closes #issue_number

- References #issue_number

### CODE REVIEW

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 1]
```diff
@@ -204,40 +200,28 @@ validate_password() {
 echo "Creating Service Principal (SP) for access to ADLA Gen2 used in Databricks mounting"
 
 # ###########################
-###if there is an hyphen retry and recreate the SP with a new password
+###if there is an hyphen in the first position replace by R
```
Comment (thesqlpro) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:203]: wording is a bit confusing in the comment
you mean replace with the letter R correct?
and should the comment be here, or in the section above (the function itself)?

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh [hunk 2]
```diff
@@ -204,40 +200,28 @@ validate_password() {
 echo "Creating Service Principal (SP) for access to ADLA Gen2 used in Databricks mounting"
```
Comment (thesqlpro) [inline e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:200]: ADLA should be ADLS

**Review by thesqlpro — COMMENTED** [review]
Just two comments to be addressed
Looks good!

File: e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh (hunk 2 above)
Comment (LiliamLeme) [reply e2e_samples/parking_sensors/scripts/deploy_infrastructure.sh:200]: ADLA was not me. It was there already when I arrived... but I fixed

**Comment by LiliamLeme** [comment]
Removed the complexity of the deploy_infra to validate the password in case there is a Hyphen
re-tested
Added a complex string on top of the makepasswrd for SQL Server so it will be a always complex when generated.

**Comment by LiliamLeme** [comment]
Removed the complexity of the deploy_infra to validate the password in case there is an Hyphen
re-tested
Added a complex string on top of the makepasswrd for SQL Server so it will be a always complex when generated.

**Pull request merged**